packages/sanity/src/desk/panes/document/timeline/timelineMenu.tsx

```diff
@@ -54,7 +57,17 @@ export function TimelineMenu({chunk, mode, placement}: TimelineMenuProps) {
     handleClose()
   }, [handleClose])
 
+  const handleGlobalKeyDown = useCallback(
```

Comment:
In addition to `Escape` – the menu should be closed also when pressing `Tab`.  Also, we should set the focus back on the button when closing the menu.

I suggest making this update to the `handleGlobalKeyDown` function:

```ts
  const handleGlobalKeyDown = useCallback(
    (event: KeyboardEvent) => {
      if (open && (isEscape(event) || event.key === 'Tab')) {
        handleClose()
        button?.focus()
      }
    },
    [handleClose, button, open]
  )

```

Reply:
Great pickup re: re-focusing!

Re: dismissing on tab, it's worth mentioning that `MenuButton` doesn't handle this behaviour either. Will add, but we should consider making that the default behaviour of `MenuButton` as well

Reply:
`MenuButton` does actually close the menu when pressing tab, see [this Arcade](https://www.sanity.io/ui/arcade?mode=jsx&jsx=eJytkDEOwyAMRXdOYXEBlo7A0KFqhx4irRxCRUjVOlIilLvXECTUrUO3%2Fx%2F%2FWzb6FHARAK57mnTYWA3o3UBG9j4Eyb4L3kUj7xgJXxk85jf5fm3IMtRXjPNxJpoiO4BbkSbpnQHhwjNbSIKyW0mOzEwqso6x1VR7IRxr%2F4whTLn6Q%2BI%2Fkfqu2l556737fVu7S2iVf9WKD1kcYv4%3D)!

Reply:
It doesn't for me 😓, neither in the studio either – let's address separately!